pypeit/spectrographs/ldt_deveny.py

```diff
@@ -0,0 +1,399 @@
+"""
+Module for LDT/DeVeny specific methods.
+
+.. include:: ../include/links.rst
+"""
+from pkg_resources import resource_filename
+
+import numpy as np
+
+from astropy.time import Time
+
+from pypeit import msgs
+from pypeit import telescopes
+from pypeit import io
+from pypeit.core import framematch
+from pypeit.spectrographs import spectrograph
+from pypeit.core import parse
+from pypeit.images import detector_container
+
+
+class LDTDeVenySpectrograph(spectrograph.Spectrograph):
+    """
+    Child to handle LDT/DeVeny specific code
+    """
+    ndet = 1
+    name = 'ldt_deveny'
+    telescope = telescopes.LDTTelescopePar()
+    camera = 'deveny'
+    comment = 'LDT DeVeny Optical Spectrograph'
+    supported = True
+
+    # Parameters equal to the PypeIt defaults, shown here for completeness
+    # pypeline = 'MultiSlit'
+
+    def get_detector_par(self, hdu, det):
+        """
+        Return metadata for the selected detector.
+
+        Args:
+            hdu (`astropy.io.fits.HDUList`_):
+                The open fits file with the raw image of interest.
+            det (:obj:`int`):
+                1-indexed detector number.
+
+        Returns:
+            :class:`~pypeit.images.detector_container.DetectorContainer`:
+            Object with the detector metadata.
+        """
+        header = hdu[0].header
+        
+        # Binning
+        binning = self.get_meta_value(self.get_headarr(hdu), 'binning')
+
+        # Detector
+        detector_dict = dict(
+            binning         = binning,
+            det             = 1,
+            dataext         = 0,        # 
+            specaxis        = 1,        # Native spectrum is along the x-axis
+            specflip        = True,     # DeVeny CCD has blue at the right
+            spatflip        = False,    # 
+            platescale      = 0.34,     # Arcsec / pixel
+            darkcurr        = 4.5,      # Electrons per hour
+            saturation      = 65535.,   # 16-bit ADC
+            nonlinear       = 1.0,      # -- Still need to measure this, close to 0.99+
+            mincounts       = -1e10,
+            numamplifiers   = 1,
+            gain            = np.atleast_1d(header['GAIN']),
+            ronoise         = np.atleast_1d(header['RDNOISE']),
+            # Data & Overscan Sections -- Edge tracing can handle slit edges
+            datasec         = np.atleast_1d('[5:512,54:2096]'),
```

Comment:
In theory, an adventurous observer could try to define a subframe, but this is a vestigial function leftover in the interface from the development of another instrument, is not included in any of our user documentation, and would not gain the observer much time since the chip reads out fairly quickly as it is.  Do you have a suggestion for a different method for defining the `datasec` and `oscansec`?  The related FITS keywords (TRIMSEC and BIASSEC) are hardwired at these values with our present controller software.

Reply:
This is fine then.  Maybe add a brief comment in the
code summarizing what you wrote above.  Thanks.